optimize/backend/src/main/java/io/camunda/optimize/license/CamundaLicense.java

```diff
@@ -0,0 +1,112 @@
+  private static final String UNKNOWN_LICENSE_TYPE = "unknown";
+  private boolean isValid;
+  private LicenseType licenseType;
+  private boolean isInitialized;
+
+  // @VisibleForTesting
+  protected CamundaLicense() {}
+
+  public CamundaLicense(final String license) {
+    initializeWithLicense(license);
+  }
+
+  public synchronized boolean isValid() {
+    return isValid;
+  }
+
+  public synchronized LicenseType getLicenseType() {
+    return licenseType;
+  }
+
+  public synchronized void initializeWithLicense(final String license) {
+    if (isInitialized) {
+      return;
+    }
+
+    if (license != null && !license.isBlank()) {
+      isValid = determineLicenseValidity(license);
+      licenseType = getLicenseTypeFromProperty(license);
+    } else {
+      isValid = false;
+      licenseType = LicenseType.UNKNOWN;
+      LOGGER.error(
+          "No license detected when one is expected. Please provide a license through the "
+              + CAMUNDA_LICENSE_ENV_VAR_KEY
+              + " environment variable.");
+    }
+
+    isInitialized = true;
+  }
+
+  /**
+   * Camunda licenses will have a license type property, fetch that out of the license and return
+   * the value
+   *
+   * <p>Self-managed mode is any other possibility. (ex, blank license, prop missing, etc)
+   */
+  private LicenseType getLicenseTypeFromProperty(final String licenseStr) {
+    try {
+      final LicenseKey licenseKey = getLicenseKey(licenseStr);
+      final String licenseType =
+          licenseKey.getProperties().getOrDefault("licenseType", UNKNOWN_LICENSE_TYPE);
+
+      if (UNKNOWN_LICENSE_TYPE.equals(licenseType)) {
+        LOGGER.error(
+            "Expected a licenseType property on the Camunda License, but none were found.");
+      }
+
+      return LicenseType.get(licenseType);
+    } catch (final InvalidLicenseException e) {
+      LOGGER.error(
+          "Expected a valid license when determining the type of license, but encountered an invalid one instead. ",
+          e);
+    } catch (final Exception e) {
+      LOGGER.error(
+          "Expected to determine the license type of the license, but the following unexpected error was encountered: ",
+          e);
+    }
+    return LicenseType.UNKNOWN;
+  }
+
+  private boolean determineLicenseValidity(final String licenseStr) {
+    try {
+      final LicenseKey license = getLicenseKey(licenseStr);
+      license.validate(); // this method logs the license status
+      return true;
+    } catch (final InvalidLicenseException e) {
+      LOGGER.error(
+          "Expected a valid license when determining license validity, but encountered an invalid one instead. ",
+          e);
+      return false;
+    } catch (final Exception e) {
+      LOGGER.error(
+          "Expected to validate a the Camunda license, but the following unexpected error was encountered: ",
+          e);
+      return false;
+    }
+  }
+
+  // @VisibleForTesting
```

Comment:
I don't think we need this as there is no test for it anyway

Reply:
Since this class is a copy of the monorepo implementation I kept the annotations to try to make the copy as close as possible to the original. I'll get these removed and add a blurb that the class will need to be deleted in the future